Physical_Engine/Modify/Transform.cs

```diff
@@ -0,0 +1,82 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System.Collections.Generic;
+using BH.oM.Dimensional;
+using BH.oM.Geometry;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.Engine.Geometry;
+using BH.oM.Physical.Elements;
+using System.Linq;
+using BH.oM.Physical.FramingProperties;
+using BH.Engine.Spatial;
+
+namespace BH.Engine.Physical
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /**** Interface Methods - IElements             ****/
+        /***************************************************/
+
+        [Description("Transforms the IFramingElement's location and orientation angle by the transform matrix. Only rigid body transformations are supported.")]
+        [Input("framingElement", "IFramingElement to transform.")]
+        [Input("transform", "Transform matrix.")]
+        [Output("transformed", "Modified IFramingElement with unchanged properties, but transformed location and orientation angle.")]
+        public static IFramingElement Transform(this IFramingElement framingElement, TransformMatrix transform)
+        {
+            if (!transform.IsRigidTransformation())
+            {
+                BH.Engine.Reflection.Compute.RecordError("Transformation failed: only rigid body transformations are currently supported.");
+                return null;
+            }
```

Comment:
Do we want to include this check in higher level methods as well? In #2339 we plan to put this check in the geometry level transform methods. It'll become redundant.
On the other hand the results of a failing check are different on those levels. Here it returns null (which is fair and logical) and in Geo_Eng it converts the input curve into `Nurbs`.

PS. tolerance passing.. 👀

Reply:
From what I understand, the checks will serve different purpose: failed check here disallows any further actions because of the discipline-specific constraints that we cannot handle, while in the Geometry_Engine it will simply change the return type (at least in the ideal world, when we have all geometry scaling/mirroring/skewing operations implemented). Therefore we need the checks on both levels I am afraid.